addons/mail/static/tests/discuss/core/channel_invitation.test.js

```diff
@@ -174,3 +174,52 @@ test("unnamed group chat should display correct name just after being invited",
         text: "You have been invited to #Jane and Mitchell Admin",
     });
 });
+
+test("invite user to self chat opens DM chat with user", async () => {
+    const pyEnv = await startServer();
+    const guestId = pyEnv["mail.guest"].create({ name: "TestGuest" });
+    const partnerId_1 = pyEnv["res.partner"].create({
+        email: "testpartner@odoo.com",
+        name: "TestPartner",
+    });
+    pyEnv["res.users"].create({ partner_id: partnerId_1 });
+    const [selfChatId] = pyEnv["discuss.channel"].create([
+        {
+            channel_member_ids: [Command.create({ partner_id: serverState.partnerId })],
+            channel_type: "chat",
+        },
+        {
+            channel_member_ids: [
+                Command.create({ partner_id: partnerId_1 }),
+                Command.create({ partner_id: serverState.partnerId }),
+            ],
+            channel_type: "group",
+        },
+        {
+            // group chat with guest as correspondent for coverage of no crash
+            channel_member_ids: [
+                Command.create({ guest_id: guestId }),
+                Command.create({ partner_id: serverState.partnerId }),
+            ],
+            channel_type: "group",
+        },
```

Comment:
We still need that for test not passing before and passing afterward.

The pick of chat conversation works regardless of other fixes, because while the detect of correspondent is wrong, the picking of chat conversation uses `channel_type === "chat"` so was not buggy.

That's why I added the group chat with guest again, added comment to justify its presence.